jwt/jwt.go

```diff
@@ -18,19 +19,26 @@ import (
 // for validating the "nbf" (Not Before) and "exp" (Expiration Time) claims.
 const DefaultLeewaySeconds = 150
 
+type Validator interface {
+	Validate(ctx context.Context, token string, expected Expected) (map[string]interface{}, error)
+	ValidateAllowMissingIatNbfExp(ctx context.Context, token string, expected Expected) (map[string]interface{}, error)
+}
+
 // Validator validates JSON Web Tokens (JWT) by providing signature
 // verification and claims set validation.
-type Validator struct {
+type validator struct {
 	keySet KeySet
 }
 
+var _ Validator = (*validator)(nil)
+
 // NewValidator returns a Validator that uses the given KeySet to verify JWT signatures.
-func NewValidator(keySet KeySet) (*Validator, error) {
+func NewValidator(keySet KeySet) (Validator, error) {
```

Comment:
I think the introduction of `Validator` isn't necessarily wrong, but I know we've tried to follow not returning interfaces with some exceptions that we think follow [generality](https://go.dev/doc/effective_go#generality). I could see the argument that this is another case for that.

I was thinking it'd be simpler and less code / api surface to make `NewValidator` a variadic function.

Something like: 
```go
func NewValidator(keySet ...KeySet) (*Validator, error) { ... }
```

I _think_ it would be roughly ~10 lines of code (tests excluded) with that approach to support multiple key sets. Happy to hear reasons why the approach here is better. Lmk!

Reply:
That is a great idea. My original intention was to minimize touching any validator logic by adding the abstraction layer with the interface. I'll have a go at the refactoring for the variadic function instead. 

I think variadic makes sense in this case since I do not expect that there will be another `Validator` implementation in the future - in which case I would lean towards the interface for extensibility.